mobile-app/cypress/integration/functional/wallet/loans/vault_detail.spec.ts

```diff
@@ -0,0 +1,137 @@
+context('Wallet - Loans - Vault Details', () => {
+  let vaultId = ''
+
+  before(function () {
```

Comment:
Wanted to discuss the usage of `beforeEach` vs `before`.

If we go with `before`, we couldn't pin point the exact test that is failing if the test case has dependency on any of the previous test.

E.g.: maybe `should check empty state` failed, and `should add loan` has dependency on it, but because the latter has dependency, it will fail too

Reply:
But taking a step back, even if we decided to use `beforeEach` and ensure each test has no dependency with each other, we still wouldn't approve a PR at least ONE test failed.

This is more to test debugging purposes, at the cost of longer e2e run time(?), and more test code(?)

Reply:
My setup would be more on an e2e flow. To test the entire integration testing that's why I preferred using `before`. Since it's an entire "test scenario" I put it into one context. If it's not, I usually put it on another context. One advantage of this is there will be less bloat of code due to similar flow. Regarding your point about debugging, I don't have a hard time debugging it since you can actually watch the video on cypress dashboard.

With this approach, it reduces e2e time, reduces test code also. But I get your point about re-structuring. I'll see how we can re-structure it differently so we can easily pinpoint failing test cases.